tests/run/opaque-immutable-array-xm.scala

```diff
@@ -0,0 +1,51 @@
+import scala.reflect.ClassTag
+object Test extends App {
+
+  import java.util.Arrays
+
+  opaque type IArray[A1] = Array[A1]
+
+  implicit object IArray {
+    inline def initialize[A](body: => Array[A]): IArray[A] = body
+    def apply[A: ClassTag](xs: A*): IArray[A] = initialize(Array(xs: _*))
+
+    // These should be inline but that does not work currently. Try again
+    // once inliner is moved to ReifyQuotes
```

Comment:
Why do these need to be inline? They are already extension methods and they take no functions or by name parameters? Wouldn’t the JIT inline these just fine for us?